ng_server/migrations/20221210_create_tables.sql

```diff
@@ -1,2 +1,2 @@
 CREATE TABLE target (block INTEGER, nonce INTEGER WITHOUT ROWID, UNIQUE(block));
-CREATE TABLE guess (player_name TEXT NOT NULL, block INTEGER REFERENCES target (block), nonce INTEGER NOT NULL, UNIQUE(player_name, block));
\ No newline at end of file
+CREATE TABLE guess (player_name TEXT NOT NULL, block INTEGER REFERENCES target (block), nonce INTEGER NOT NULL);
```

Comment:
Rather than removing the unique constraint a better fix is to add a new constraint that prevents duplicate player names prior to selecting the target block. The app could use some better error handling in the UI to let the user know their guess wasn't accepted because they gave a non-unique name, but at least the db can't get into an invalid state with this new constraint and will always be able to update without errors when the block is found.

```suggestion
CREATE TABLE guess (player_name TEXT NOT NULL, block INTEGER REFERENCES target (block), nonce INTEGER NOT NULL, UNIQUE(player_name, block));
CREATE UNIQUE INDEX idx_guess_null_block ON guess(player_name) WHERE block IS NULL;
```